packages/amplication-git-utils/src/git/git.service.ts

```diff
@@ -89,39 +99,213 @@ export class GitClientService {
     }
 
     if (pullRequestMode === EnumPullRequestMode.Accumulative) {
-      await this.provider.createBranchIfNotExists({
+      const gitClient = new GitClient();
+      const cloneFolder = process.env.CLONES_FOLDER;
+      if (!cloneFolder) {
+        throw new MissingEnvParam("CLONES_FOLDER");
+      }
+
+      const randomUUID = v4();
+
+      const cloneUrl = this.provider.getCloneUrl({
+        owner,
+        repositoryName,
+      });
+
+      const cloneDir = normalize(
+        join(cloneFolder, this.provider.name, owner, repositoryName, randomUUID)
+      );
+
+      await gitClient.clone(cloneUrl, cloneDir);
+
+      await this.restoreAmplicationBranchIfNotExists({
         owner,
         repositoryName,
         branchName,
-      }),
-        await this.provider.createCommit({
+        gitClient,
+      });
+
+      const diffFolder = normalize(
+        join(
+          `.amplication/diffs`,
+          this.provider.name,
           owner,
+
           repositoryName,
-          commitMessage,
-          branchName,
-          files: preparedFiles,
+          randomUUID
+        )
+      );
+
+      const { diff } = await this.preCommitProcess({
+        branchName,
+        gitClient,
+        owner,
+        repositoryName,
+      });
+      let fullDiffPath = "";
+      if (diff) {
+        await mkdir(diffFolder, { recursive: true });
+        const diffPath = join(diffFolder, "diff.patch");
```

Comment:
maybe it is worth to merge this if statement with the following one since the `this.provider.createCommit` function in the middle is not affected by the results of this block.